rfcs/text/0003_handler_interface.md

```diff
@@ -0,0 +1,244 @@
+- Start Date: 2019-05-11
+- RFC PR: (leave this empty)
+- Kibana Issue: (leave this empty)
+
+# Summary
+
+Handlers are asynchronous functions that get passed a context object including
+capabilities and observables that can be provided by the service that owns the
+handler (aka service owner) or other services that are not necessarily known to
+the service owner.
+
+# Basic example
+
+If the proposal involves a new or changed API, include a basic code example.
+Omit this section if it's not applicable.
+
+```js
+http.router.route({
+  method: 'GET',
+  path: '/saved_object/:id',
+  // routeHandler implements the "handler" interface
+  async routeHandler(context, request) {
+    return context.core.savedObjects.find(request.params.id);
```

Comment:
what is the interface on context? { core, plugins } ? where core is always defined, plugins is dynamic with optional keys?
because lower `context` is an object with flatten keys

Reply:
Initially, I was considering not having a special Core key on this interface, but I forgot to change this example. After thinking about this more, I think it's good to have Core specified as a top-level key explicitly that is always known to be present in the type information.